pkg/scheduler/placement/placement.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2019 Cloudera, Inc.  All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+    "fmt"
+    "github.com/cloudera/yunikorn-core/pkg/cache"
+    "github.com/cloudera/yunikorn-core/pkg/common/configs"
+    "github.com/cloudera/yunikorn-core/pkg/log"
+    "go.uber.org/zap"
+    "sync"
+)
+
+type Manager struct {
+    name        string
+    info        *cache.PartitionInfo
+    rules       []rule
+    initialised bool
```

Comment:
We need it because we can not be sure that the manager is always initialised. The scheduler can run with or without the placement rules configured.
Then after a reload of the configuration it can trigger a rule update and the manager which already exists can then be used.

Reply:
We did need a lock in the `IsInitialised` method, adding this in the update

Reply:
Instead of doing that, maybe we can always have the placement manager initiated. If no rule is configured, we can use a default `EmptyRule` which is basically a noop when place applications. Will this help?

Reply:
That option has been added as a TODO in the scheduling context already. It requires a bit more refactoring and thought. The way we currently have it is backward compatible.

When using a placement rule in all cases, even with a noop rule, we start enforcing ACLs. This will break all existing configs which is even at this point a no go.

Reply:
opened #33 to implement this